server/util.go

```diff
@@ -316,7 +316,7 @@ func getURLsAsString(urls []*url.URL) []string {
 // copyBytes make a new slice of the same size than `src` and copy its content.
 // If `src` is nil, then this returns `nil`
 func copyBytes(src []byte) []byte {
-	if src == nil {
```

Comment:
Would have maintained the original semantic: you pass nil, you get nil back, but if you pass empty slice, you get an empty slice. Hopefully you are not breaking code that would have depended on that...

Reply:
empty slice should not cause new malloc imo, so when I saw that I thought it was wrong.

Reply:
Since if interior could be empty but non-nil.

Reply:
When I wrote this helper, I added the comment that if src is nil, then result is nil, but otherwise the new slice is same size than original, which, I agree could mean allocating a slice of size 0, but that meant a matching source/destination in term of nil vs not nil. But I think that's fine... although updating the doc then would match better the new behavior.

Reply:
Updated comment.